simtools/data_model/metadata_model.py

```diff
@@ -1,281 +1,178 @@
-        "START": {"type": "datetime", "required": False, "default": None},
-        "SOFTWARE": {
-            "NAME": {"type": "str", "required": True, "default": None},
-            "VERSION": {"type": "str", "required": True, "default": None},
-        },
-    }
-
-
-def top_level_reference_schema():
+def top_level_reference_schema(schema_file=None):
     """
     Reference schema following the CTA Top-Level Data Model.
 
     This metadata schema is used for simtools data products.
 
+    Parameters
+    ----------
+    schema_file: str
+        Schema file (jsonschema format) used for validation
+
     Returns
     -------
     dict with reference schema
 
 
     """
 
-    _ref_schema = {
-        "CTA": {
-            "REFERENCE": _category_reference(),
-            "CONTACT": _category_contact(),
-            "PRODUCT": _category_product(),
-            "INSTRUMENT": _category_instrument(),
-            "PROCESS": _category_process(),
-            "ACTIVITY": _category_activity(),
-            "CONTEXT": {
-                "SIM": {
-                    "ASSOCIATION": [
-                        _category_instrument(),
-                    ],
-                    "DOCUMENT": [
-                        _category_document(),
-                    ],
-                },
-            },
-        }
-    }
-    _ref_schema = _metadata_dict_with_defaults(_ref_schema)
-    return _remove_empty_lists(_ref_schema)
-
-
-def metadata_input_reference_schema():
-    """
-    Reference data model scheme for input metadata.
-    Describes metadata provided for input to simtools applications.
+    schema = load_schema(schema_file)
+    return fill_defaults(schema["definitions"]["CTA"])
 
-    Returns
-    -------
-    dict with input reference schema
-
-    """
 
-    return {
-        "REFERENCE": _category_reference(),
-        "CONTACT": _category_contact(),
-        "PRODUCT": _category_product(),
-        "INSTRUMENT": _category_instrument(),
-        "PROCESS": _category_process(),
-        "CONTEXT": _category_context(),
-    }
-
-
-def metadata_input_reference_document_list(schema_list):
+def resolve_references(yaml_data):
     """
-    Reference model data for input metata data of type documentlist or instrumentlist
+    Resolve references in yaml data and expand the
+    received dictionary accordingly.
 
     Parameters
     ----------
-    schema_list: str
-        List type to be returned (e.g., instrumentlist or documentlist)
+    yaml_data: dict
+        Dictionary with yaml data.
 
     Returns
     -------
-    dict with input reference schema for documentlist or instrumentlist
+    dict
+        Dictionary with resolved references.
 
     """
-    if schema_list.lower() == "instrumentlist":
-        return _category_instrument()
-    if schema_list.lower() == "documentlist":
-        return _category_document()
 
-    msg = f"Invalid schema list: {schema_list}"
-    raise InvalidSchemaList(msg)
+    def expand_ref(ref):
```

Comment:
This is a 'sub function' (not sure what it is called in python) - a function inside a function. This should be covered by `test_resolve_references`.

Reply:
Hmm, I am not sure that is a good practice. The sub function might be used in a second function in the future or it could even be used independently, in which case we miss the test. If we want to speed up the test, we could use a mocker, but in this case the time to run the test is not an issue.

Reply:
This is standard python practice and we can find it a couple of times in simtools.